wagtail/contrib/postgres_search/backend.py

```diff
@@ -92,25 +95,25 @@ def id(self):
     @cached_property
     def body(self):
         """
-        Returns all values to index as "body". This is the value of all SearchFields that have partial_match=False
+        Returns all values to index as "body". This is the value of all SearchFields
         """
         texts = []
         for field in self.search_fields:
             for current_field, boost, value in self.prepare_field(self.obj, field):
-                if isinstance(current_field, SearchField) and not current_field.partial_match:
+                if isinstance(current_field, SearchField):
                     texts.append((value, boost))
 
         return self.as_vector(texts)
 
     @cached_property
     def autocomplete(self):
         """
-        Returns all values to index as "body". This is the value of all SearchFields that have partial_match=True
+        Returns all values to index as "autocomplete". This is the value of all AutocompleteFields
         """
         texts = []
         for field in self.search_fields:
             for current_field, boost, value in self.prepare_field(self.obj, field):
-                if isinstance(current_field, SearchField) and current_field.partial_match:
```

Comment:
I'm okay with us breaking backwards compatibility on `AutocompleteField`, but isn't this also breaking the behaviour of `SearchField(partial_match=True)`, which _is_ documented? Shouldn't we be indexing both `AutocompleteField` and `SearchField(partial_match=True)` into the autocomplete index, until the latter is formally deprecated?

Reply:
The autocomplete field is only used on its own when using the undocumented `autocomplete()` API. When users use the `.search()` API it's just merged back into `body` at runtime and treated no differently.

Reply:
Ahh, I see... I missed the fact that the Postgres backend has never supported the current documented functionality of `partial_match=True` (i.e. returning partial matches on `search()`), so any references to `partial_match` that do exist in the code are only there to support the previous iteration of `autocomplete()`, and are fair game to be removed.

Having said that, it looks like there's one remaining reference to `partial_match` in the backend after this PR:

https://github.com/kaedroho/wagtail/blob/d536e73baf017898b3bd6570d4221d227709ea60/wagtail/contrib/postgres_search/backend.py#L472-L473

I guess this should be updated to iterate over `self.queryset.model.get_autocomplete_search_fields()` instead?

Reply:
Ahh, yep that's right will fix!

I think the tests have been working because the title field in the test model has both `partial_match=True` and is an `AutocompleteField`

Reply:
Proposed fix in a329870a8e18fa7faea04139ee44b9a250cf04d5.

It turns out my "why can't we use boost on AutocompleteFields?" tangent is somewhat relevant here after all - having patched `PostgresAutocompleteQueryCompiler` to build `self.search_fields` as a lookup of `AutocompleteField`s rather than `SearchField`s, it turns out that `get_fields_vector` also needs patching to hard-code a weight of 'D', rather than calling `get_weight(search_field.boost)` (which is not currently valid on AutocompleteField).

If AutocompleteField supported a `boost` field - and I can't see a logical reason why it shouldn't - we could get away without overriding this. Happy to leave that for a future PR though.